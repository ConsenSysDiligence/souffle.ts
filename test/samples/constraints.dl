// https://souffle-lang.github.io/constraints

.decl stringTable(t:symbol)
.decl substringTable(t:symbol)
.decl outputData(substrs:symbol, str:symbol)
.output outputData
outputData(x,y) :- substringTable(x), stringTable(y), contains(x,y).
stringTable("aaaa").
stringTable("abba").
stringTable("bcab").
stringTable("bdab").
substringTable("a").
substringTable("ab").
substringTable("cab").

.decl inputData(t:symbol)
.decl outputData2(t:symbol)
.output outputData2
outputData2(x) :- inputData(x), match("a.*",x).
inputData("aaaa").
inputData("abba").
inputData("bcab").
inputData("bdab").
