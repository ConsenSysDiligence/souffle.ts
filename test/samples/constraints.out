/// inputData
===============
t
aaaa
abba
bcab
bdab
/// outputData
===============
substrs str
a    aaaa
a    abba
a    bcab
a    bdab
ab    abba
ab    bcab
ab    bdab
cab    bcab
/// outputData2
===============
t
aaaa
abba
/// stringTable
===============
t
aaaa
abba
bcab
bdab
/// substringTable
===============
t
a
ab
cab